src/services/highcharts-factory/highcharts-factory.svc.coffee

```diff
@@ -43,30 +28,33 @@ angular
             { type: 'year', count: 1, text: '1y' },
             { type: 'all', text: 'All' }
           ]
-          selected: (if _.get(xAxisOptions, 'min') then null else 0)
 
   todayUTC = moment().startOf('day').add(moment().utcOffset(), 'minutes')
 
   class Chart
-    constructor: (@id, @data = {}, @options = {})->
-      @_template = templates[@options.chartType]
+    constructor: (@id, @series = {}, @settings = {})->
+      @_template = templates[@settings.chartType]
+      @populateChartOptions(@series, @settings)
       return
 
-    render: (data, options)->
-      @data = data if _.isObject(data)
-      angular.extend(@options, options)
-      chartConfig = angular.merge({}, @template(), @formatters(), @todayMarker())
-      if _.isEmpty(@hc)
-        @hc = Highcharts.stockChart(@id, chartConfig)
-      else
-        @hc.update(chartConfig)
+    populateChartOptions: (series, settings) ->
+      # Every chart will have these options. The rest of the options are created in the setter methods.
+      template = @_template
```

Comment:
If you move the contents of this `populateChartOptions` method to just the constructor block, as @_template is not being accessed anywhere, we can remove it from the `this` scope, and just use 
```
template = templates[@settings.chartType]
angular.merge({}, templates ... ... . .) 
```